### ISSUE TITLE
Label PODs instead of services

### ISSUE THREAD

**Issue by leszko** [hazelcast/hazelcast-kubernetes#115]
Filtering Hazelcast members by labeling services is weird. The more straightforward approach would be to just label PODs and make the labeling services deprecated.

Related to : #113

**Comment by bitsofinfo** [comment]
agreed, please let it just select on pod labels. Or at least an option too!

**Comment by henrikarro** [comment]
Maybe I'm missing something, but I think we still need a way to support smart clients outside the Kubernetes cluster, and for this use-case, `service-label-name` and `service-label-value` are perfect.

If you have Hazelcast embedded in a server, you will have other services in the same pod, not just Hazelcast, so it seems to me you need a way to tell the Hazelcast client which services to connect to.

**Comment by leszko** [comment]
@henrikarro , would it change anything for you if you label Pods which contain Hazelcast instead of the services? That's I'm  trying to understand.

Because to me, labeling Pods should cover all use cases and is more straightforward. You label Pods and you discover Pods, so may not even create a service if not needed.

**Comment by henrikarro** [comment]
@leszko: I'm working on a system where we have server PODs that both publish an API (let's say on port 8080) and also give access to an embedded Hazelcast cluster (let's say on port 5701). We have clients outside the Kubernetes cluster that need to access both the API and Hazelcast. To enable Hazelcast "smart clients", we create a NodePort service-per-pod to let the clients access any of the Hazelcast nodes in the cluster.

In this case, I don't see how labeling PODs would work:
* For clients outside the Kubernetes cluster to be able to reach the PODs, you need to create services.
* PODs can be reachable through multiple services so I guess it's difficult for the Hazelcast client to know which service to connect to.

In short, I think `service-name` and `service-label-name`/`service-label-value` would still be useful for Hazelcast clients outside the Kubernetes cluster.

**Comment by leszko** [comment]
@henrikarro I see your use case. You could still label PODs with Hazelcast (and your application). However, I understand you may not want to to add a Hazelcast-specific label to the POD which contains the application.

Ok, let's keep both Service Label and Pod Label.

**Comment by bitsofinfo** [comment]
Yes, please add the option for pods, and keep the existing method for services

**Comment by hasancelik** [comment]
@bitsofinfo @henrikarro With [1.5.2](https://github.com/hazelcast/hazelcast-kubernetes/releases/tag/v1.5.2) release, the discovery plugin can discover Hazelcast members via pod labels. 

Thanks for your feedbacks and thoughts.

**Comment by Dineshseervi** [comment]
These two are very help full to over come any connection issue we face while clustering like HTT or not able to form cluster due to lots of pod to scan .Plz try service-name or below that will help to use Hazelcast in your application 
-->
join.getKubernetesConfig().setProperty("service-label-name", "label-name-you-want-to-use-from-pod");
join.getKubernetesConfig().setProperty("service-label-value", "label-value-for-lable-name");
or  (dont use service label and pod label together)
join.getKubernetesConfig().setProperty("pod-laber-name", "label-name-you-want-to-use-from-service");
join.getKubernetesConfig().setProperty("pod-laber-value", "label-value-for-lable-name");
